plugin-server/src/main/ingestion-queues/batch-processing/each-batch-ingestion.ts

```diff
@@ -255,7 +255,7 @@ async function emitToOverflow(queue: IngestionConsumer, kafkaMessages: Message[]
             queue.pluginsServer.kafkaProducer.produce({
                 topic: KAFKA_EVENTS_PLUGIN_INGESTION_OVERFLOW,
                 value: message.value,
-                key: null, // No locality guarantees in overflow
+                key: message.key,
```

Comment:
`message.key` can be empty if capture-side overflow detection triggers.

As we want to evaluate impact and probably don't want to invest a lot for now, I'm fine with:
  - checking how much capture-side detection triggers (on both new and old capture) vs plugin-server-side
  - disable capture-side detection on both captures for now while we evalutate this

Last monday's incident has shown us that we can read & unmarshall really fast (1.6M/minute with 8 historical pods dropping on token), so capture-side might not be really necessary.

Reply:
Going to hold off on merging this for a bit — I had assumed there was already a method to turn off overflow routing wholesale on both captures but it doesn't look like that was a valid assumption for me to make. Seems like it'd make sense to take care of that first.

Reply:
It can be [turned off in python](https://github.com/PostHog/posthog/blob/e5d29fa4c0f58bf433cf15523e12f5d3fe74db0f/posthog/settings/ingestion.py#L25-L29), but you are right that it cannot be turned off in rust yet. I'd put very high thresholds in the rust config while working on a PR to add a boolean config to completely disable it.
BTW, this capture-side detection is kind of a scalability time bomb as its memory usage is O(active distinct_id), so it needs to eventually be phased out anyway.

Reply:
> It can be [turned off in python](https://github.com/PostHog/posthog/blob/e5d29fa4c0f58bf433cf15523e12f5d3fe74db0f/posthog/settings/ingestion.py#L25-L29), but you are right that it cannot be turned off in rust yet.

Ah, I was thinking we'd want to bypass [this entire conditional](https://github.com/PostHog/posthog/blob/a4bba2f36ca80038582be6db55261959836bb2db/posthog/api/capture.py#L629-L630), but maybe that's overkill.

> It can be [turned off in python](https://github.com/PostHog/posthog/blob/e5d29fa4c0f58bf433cf15523e12f5d3fe74db0f/posthog/settings/ingestion.py#L25-L29), but you are right that it cannot be turned off in rust yet. I'd put very high thresholds in the rust config while working on a PR to add a boolean config to completely disable it.
BTW, this capture-side detection is kind of a scalability time bomb as its memory usage is O(active distinct_id), so it needs to eventually be phased out anyway.

Good point, I hadn't really considered that — thanks for mentioning it.

Reply:
Good call, we'd need to skip the check against LIKELY_ANONYMOUS_IDS too

Reply:
Ability to turn off random partitioning completely in old capture is here: #21168 (My initial thought was to keep that PR separate, but in retrospect, I suppose this could have been part of this change too.)